pygmo/docstrings.cpp

```diff
@@ -3601,4 +3596,416 @@ inserted via :func:`~pygmo.core.archipelago.push_back()`).
+   description of each supported solver.
+
+Args:
+    solver (``str``): the name of the NLopt algorithm that will be used by this :class:`~pygmo.core.nlopt` object
+
+Raises:
+    RuntimeError: if the NLopt version is not at least 2
+    ValueError: if *solver* is not one of the allowed algorithm names
+    unspecified: any exception thrown by failures at the intersection between C++ and Python (e.g.,
+      type conversion errors, mismatched function signatures, etc.)
+
+Examples:
+    >>> from pygmo import *
+    >>> nl = nlopt('slsqp')
+    >>> nl.xtol_rel = 1E-6 # Change the default value of the xtol_rel stopping criterion
+    >>> nl.xtol_rel # doctest: +SKIP
+    1E-6
+    >>> algo = algorithm(nl)
+    >>> algo.set_verbosity(1)
+    >>> prob = problem(luksan_vlcek1(20))
+    >>> prob.c_tol = [1E-6] * 18 # Set constraints tolerance to 1E-6
+    >>> pop = population(prob, 20)
+    >>> pop = algo.evolve(pop) # doctest: +SKIP
+       fevals:       fitness:      violated:    viol. norm:
+             1        95959.4             18        538.227 i
+             2        89282.7             18        5177.42 i
+             3          75580             18        464.206 i
+             4          75580             18        464.206 i
+             5        77737.6             18        1095.94 i
+             6          41162             18        350.446 i
+             7          41162             18        350.446 i
+             8          67881             18        362.454 i
+             9        30502.2             18        249.762 i
+            10        30502.2             18        249.762 i
+            11        7266.73             18        95.5946 i
+            12         4510.3             18        42.2385 i
+            13        2400.66             18        35.2507 i
+            14        34051.9             18        749.355 i
+            15        1657.41             18        32.1575 i
+            16        1657.41             18        32.1575 i
+            17        1564.44             18        12.5042 i
+            18        275.987             14        6.22676 i
+            19        232.765             12         12.442 i
+            20        161.892             15        4.00744 i
+            21        161.892             15        4.00744 i
+            22        17.6821             11        1.78909 i
+            23        7.71103              5       0.130386 i
+            24        6.24758              4     0.00736759 i
+            25        6.23325              1    5.12547e-05 i
+            26         6.2325              0              0
+            27        6.23246              0              0
+            28        6.23246              0              0
+            29        6.23246              0              0
+            30        6.23246              0              0
+    <BLANKLINE>
```

Comment:
no need for <BLANKLINE> since you skip the test I guess .. on the other hand is clearer like this and consistent ... probably leave it ..